core/dbt/parser/manifest.py

```diff
@@ -125,6 +126,36 @@
 PERF_INFO_FILE_NAME = "perf_info.json"
 
 
+def extended_mashumaro_encoder(data):
+    return msgpack.packb(data, default=extended_msgpack_encoder, use_bin_type=True)
+
+
+def extended_msgpack_encoder(obj):
+    if type(obj) is datetime.date:
```

Comment:
if deprecation_date always be a datetime - do we strictly need this case for the deprecation_date addition? or is it just provided here in the event that we do want to serialize a date object in the future

Reply:
Piggy-backing off your question @MichelleArk, here are some of my beliefs:

The happiest path is if `deprecation_date` is converted to an "aware" datetime in Python that can compare with another "aware" datetime that represents "now" for dbt purposes.

That way, the user can specify an unambiguous timestamp in [RFC 3339](https://dbeatty10.github.io/rfc3339-iso8601/) format like `2023-05-09 11:15:00-0600`, and it can compare nicely to the current system time like `2023-05-09 17:13:58Z`.

Anything other than two aware datetimes is likely to be confusing to dbt users and/or dbt-core developers.

Reply:
@MichelleArk If the user puts a date into the deprecation_date field, such as '2023-05-09', the YAML parser produces a raw parsed object which contains a date, rather than a date-time. That raw parser output is one of the things that gets put in the manifest, which is why I have included date. Writing and reading the manifest fails, otherwise.

Reply:
@dbeatty10 I agree, and will work on getting dates converted to timezone-aware as early as possible in the parsing process.